Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Components/Objects/GetObjectProperties.cs

```diff
@@ -60,12 +61,29 @@ protected override void SolveInstance(IGH_DataAccess da)
     }
     else
     {
-      SpeckleGeometryWrapperGoo objectWrapperGoo = new();
-      da.GetData(0, ref objectWrapperGoo);
+      IGH_Goo? inputObject = null;
+      SpecklePropertyGroupGoo? properties = null;
 
-      // flatten object properties, if any
-      SpecklePropertyGroupGoo properties = objectWrapperGoo.Value.Properties;
-      if (properties.Value.Count == 0)
+      if (da.GetData(0, ref inputObject))
+      {
+        // Handle both Geometry and DataObject wrappers using existing extension method
+        if (inputObject?.ToSpeckleGeometryWrapper() is SpeckleGeometryWrapper geoWrapper)
+        {
+          properties = geoWrapper.Properties;
+        }
+        // Handle DataObjects directly
+        else if (inputObject is SpeckleDataObjectWrapperGoo dataObjectGoo)
```

Comment:
I would start with the dataobject case first (the tospecklegeometrywrapper will cast)

Reply:
Swapped the order 👍🏼 [refactor: pr review comments](https://github.com/specklesystems/speckle-sharp-connectors/pull/978/commits/d1feb4583518c9f576f18b6d1386e387069b924e)